mypyc/test-data/run-misc.test

```diff
@@ -875,6 +875,89 @@ assert call_all(mixed_110) == 1
 assert call_any_nested([[1, 1, 1], [1, 1], []]) == 1
 assert call_any_nested([[1, 1, 1], [0, 1], []]) == 0
 
+
+[case testSum]
+[file driver.py]
+print('test sums of numbers')
+print(sum((x for x in [0, 1, 2]), 0))
+print(sum((x for x in [0.0, 1, 2]), 0))
+print(sum((x for x in [1, 1j])))
+print(sum((x**2 for x in [0, 1.1, 8.3, 5.3])))
+
+print('test cases where first arg to sum() is a Generator of Callables')
+print(sum((lambda x: x == 0)(x) for x in []))
+print(sum((lambda x: x == 0)(x) for x in [0]))
+print(sum((lambda x: x == 0)(x) for x in [0, 0, 0]))
+print(sum((lambda x: x == 0)(x) for x in [0, 1, 0]))
+print(sum((lambda x: x % 2 == 0)(x) for x in range(2**10)))
+print(sum(((lambda x: x % 2 == 0)(x) for x in range(2**10)), 1))
+print(sum(((lambda x: x % 2 == 0)(x) for x in range(2**10)), start=1))
+
+print('test cases where first arg to sum() is a Generator of Comparison')
+print(sum(x == 0 for x in []))
+print(sum(x == 0 for x in [0]))
+print(sum(x == 0 for x in [0, 0, 0]))
+print(sum(x == 0 for x in [0, 1, 0]))
+print(sum(x % 2 == 0 for x in range(2**10)))
+print(sum((x % 2 == 0 for x in range(2**10)), 1))
+print(sum((x % 2 == 0 for x in range(2**10)), start=1))
+
+print('test multi-layered generator')
+print(sum(i + j == 0 for i, j in zip([0, 0, 0], [0, 1, 0])))
+
+print('test misc cases')
+print(sum((x == 0 for x in [0, 1]), 0 + 1j))
+a = 1
+print(sum((x == 0 for x in [0, 1]), a))
+
+# misc cases we do optimize (note, according to sum's helptext, we don't need to support non-numeric
+# cases, but CPython and mypyc both do anyway)
+print('test misc cases we do optimize')
+print(sum((x == 0 for x in [0, 1]), 0.5))
+print(sum((x == 0 for x in [0, 1]), 1j))
+print(sum(c == 'd' for c in 'abcdd'))
+print(sum((c == 'd' for c in 'abcdd'), 1))
+print(sum((c == 'd' for c in 'abcdd'), start=1))
```

Comment:
We need to drop the `start`, since it isn't supported on all the pythons we need to support. (We could add a test to `run-python38.test`, but it hardly seems worth it for this.)